virttest/utils_libvirt/libvirt_network.py

```diff
@@ -60,24 +63,42 @@ def create_or_del_network(net_dict, is_del=False, remote_args=None):
 
 def check_established(params):
     """
-    Parses netstat output for established connection on remote
+    Parses netstat output for established connection
+    on remote or local
 
     :param params: the parameters used
+    :return: str, the port or None
     """
     port_to_check = params.get("port_to_check", "4915")
-
+    check_local = 'yes' == params.get("check_local_port", "no")
+    return_port = "yes" == params.get("return_port", "no")
     ipv6_config = "yes" == params.get("ipv6_config", "no")
     if ipv6_config:
         server_ip = params.get("ipv6_addr_des", "")[:17]
         client_ip = params.get("ipv6_addr_src", "")[:17]
     else:
-        server_ip = params.get("server_ip")
-        client_ip = params.get("client_ip")
+        server_ip = params.get("server_ip", params.get("remote_ip"))
+        client_ip = params.get("client_ip", params.get("local_ip"))
 
     cmd = "netstat -tunap|grep %s" % port_to_check
-    exp_msg = r".*%s:%s.*%s.*ESTABLISHED.*qemu-kvm.*" % (server_ip,
-                                                         port_to_check,
-                                                         client_ip)
+    if check_local:
+        exp_msg = r".*%s:\d*.*%s:%s.*ESTABLISHED.*qemu-kvm.*" % (client_ip,
+                                                                 server_ip,
+                                                                 port_to_check)
+        cmdRes = process.run(cmd, shell=True)
+    else:
+        exp_msg = r".*%s:%s.*%s.*ESTABLISHED.*qemu-kvm.*" % (server_ip,
+                                                             port_to_check,
+                                                             client_ip)
+        cmdRes = remote.run_remote_cmd(cmd, params)
 
-    cmdRes = remote.run_remote_cmd(cmd, params)
     libvirt.check_result(cmdRes, expected_match=exp_msg)
+
+    if return_port:
```

Comment:
What about returning the port directly rather than checking the value of 'return_port'?

Reply:
Updated